agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -1836,6 +1836,8 @@ bool backhaul_manager::handle_1905_discovery_query(ieee1905_1::CmduMessageRx &cm
         for (const auto &vap : vaps_list.second.vaps) {
             if (vap.mac == network_utils::ZERO_MAC)
                 continue;
+            if (vap.ssid[0] == '\0')
```

Comment:
Is it guaranteed that empty ssid means that there is no beacon?
Why not instead add another member to the VAP structure which states the VAP state?
Yes, this would mean we have to implement this in all BWL backends, but isn't it cleaner?

Reply:
Yeah, I didn't want to change too much...

The situation is a bit schizophrenic. The bwl interface uses a map vap_index->vap_info, so it kind of assumes that it will only contain "valid" vaps. But then the messages use sVapsList which has a fixed size IFACE_TOTAL_VAPS, which kind of assumes a pre-allocated number of VAPs. And the bwl implementations also all have a fixed number of VAPs they support, bounded by IFACE_VAP_ID_MAX (which is based on IFACE_TOTAL_VAPS).

In my book, it's a mess :-) So I didn't want to touch it too much.